### PR TITLE
[DA-228] Remove hardcoded application version, add property loading 

### PR DESCRIPTION
**Pull request by jomrazek**
I can't find out right settings of resource plugin to filter testsuite/src/test/rest/v-1.0/expectedResponse/index.html - i will look at it

I created https://docs.engineering.redhat.com/display/JP/Changing+of+REST+API+version so you know all places where you currently need to change version. We're using constants in app and these constants are used in annotations so I think the best state we can get is changing rest versions on 2 places in constants(for app) and in parent pom(for filter) 

If you know how to read property to annotation let me know.

### CODE REVIEW

File: bc-rest/src/main/webapp/WEB-INF/web.xml [hunk 1]
```diff
@@ -7,11 +7,11 @@
         <servlet-class>io.swagger.jaxrs.config.DefaultJaxrsConfig</servlet-class>
         <init-param>
             <param-name>api.version</param-name>
-            <param-value>0.3.0</param-value>
+            <param-value>${version.bc.rest}.0</param-value>
```
Comment (thescouser89) [inline bc-rest/src/main/webapp/WEB-INF/web.xml]: Do we have to set the version as <>.<>.<>, or can we just use <>.<>  ? I have a feeling the trailing '.0' might bite us  sometimes, but I don't remember anymore what does 'api.version' do.

File: reports-rest/src/main/webapp/WEB-INF/web.xml [hunk 2]
```diff
@@ -7,11 +7,11 @@
         <servlet-class>io.swagger.jaxrs.config.DefaultJaxrsConfig</servlet-class>
         <init-param>
             <param-name>api.version</param-name>
-            <param-value>0.3.0</param-value>
+            <param-value>${version.reports.rest}.0</param-value>
```
Comment (thescouser89) [inline reports-rest/src/main/webapp/WEB-INF/web.xml]: Same here :smiley:

**Comment by jbartece** [comment]
I think that Dustin is right, that it should be ideally X.Y.Z, but I checked it and I don't see any easy solution for this except using 4 properties in Maven. So, I'd vote for left it as it is and use only X.Y for the REST API and when we will make any backward incompatible changes, we'll increase the number

**Comment by jbartece** [comment]
No objections, I'm merging

**Pull request merged**